### PR TITLE
Issue #68: Set up and configure Doorkeeper to provide OAuth with the password grant type

### PR DESCRIPTION
**Pull request by joshmcarthur**
This pull request adds the ability to authenticate against a very simple API endpoint I have added to access the logged-in-user's basic information, using the OAuth 2 password grant type. This means that you `POST` the user's normal sign in credentials (email, password) to the application, and receive an expiring access token in return. This access token can subsequently be used to access the API on behalf of the user.

For now, I have predicted that really only the "official" apps (iOS, Android) will be using the OAuth access, which is why the grant type is limited to password (it is not the most secure option, as it involves collecting and transmitting the password from the client, but it is extremely easy to get going with, works great for apps, and doesn't require the design and UX work of a normal "Do you want application X to have access to data Y and Z?" flow). I have also restricted the forms that Doorkeeper provides to create an application, and instead written a very simple rake task that will collect the necessary information and issue a client ID and secret in return.

The best way to famliarise with how the OAuth process actually works is to check out `spec/requests/oauth_authorization_spec.rb`, which has a transaction of a client authorising on behalf of a user.

I _have_ added a single API controller, however this is mostly in order to give testers at least a single endpoint to hit that requires access token authorisation. It works, but in the long term I would recommend adding a gem that takes care of all the dull heavy lifting of creating an API, such as [`grape`](https://github.com/ruby-grape/grape) or [`jsonapi-resources`](https://github.com/cerebris/jsonapi-resources). 

In terms of overcommit, I would like to ask that the YAML line length be raised to 100 or 120 characters. I made some changes to break long lines onto multiple lines, but 80 characters in an indented file format does not leave much breathing room.

After merging, I believe this PR will close issue #68

### CODE REVIEW

**Comment by Br3nda** [comment]
Trying to retrigger codeclimate...

File: Gemfile.lock [hunk 1]
```diff
@@ -59,7 +59,7 @@ GEM
     capybara-screenshot (1.0.17)
       capybara (>= 1.0, < 3)
       launchy
-    chartkick (2.2.4)
+    chartkick (2.2.3)
```
Comment (Br3nda) [inline Gemfile.lock]: these downgrades are a bit curious

**Comment by Br3nda** [comment]
Yaml lint config added in #307..

File: Gemfile.lock (hunk 1 above)
Comment (joshmcarthur) [inline Gemfile.lock]: @Br3nda I think they originate from this branch being started in May and not updated from upstream until it was feature complete. I'll try and reset to the Gemfile.lock from master and hopefully they will come right.

**Comment by Br3nda** [comment]
I need a bit more time to test this... 
It's going to conflict with #303  when that merges.

**Comment by Br3nda** [comment]
Works great! thanks @joshmcarthur

**Pull request merged**